Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -0,0 +1,240 @@
+    int sizeArray=numAtoms*numAtoms;
+    double *Geodesic = new double[sizeArray];
+    for (int i=0; i<sizeArray;i++) {
+      if (dist[i]==lag) Geodesic[i]=1;
+      else  Geodesic[i]=0;
+    }
+
+    return Geodesic;
+}
+
+
+double* get3DautocorrelationDesc(double* dist3D, double* dist, int numAtoms, const ROMol& mol) {
+
+   Map<MatrixXd> dm(dist3D, numAtoms,numAtoms);
+
+   double *w= new double[80];
+
+   std::vector<double> wp= moldata3D.GetRelativePol(mol);
+
+   VectorXd Wp = getEigenVect(wp);
+
+   std::vector<double> wm= moldata3D.GetRelativeMW(mol);
+
+   VectorXd Wm = getEigenVect(wm);
+
+   std::vector<double> wi= moldata3D.GetRelativeIonPol(mol);
+
+   VectorXd Wi = getEigenVect(wi);
+
+   std::vector<double> wv= moldata3D.GetRelativeVdW(mol);
+
+   VectorXd Wv = getEigenVect(wv);
+
+   std::vector<double> we= moldata3D.GetRelativeENeg(mol);
+
+   VectorXd We = getEigenVect(we);
+
+   std::vector<double>  wu = moldata3D.GetUn(numAtoms);
+
+   VectorXd Wu = getEigenVect(wu);
+
+   std::vector<double>  ws =  moldata3D.GetIState(mol);
+
+   VectorXd Ws = getEigenVect(ws);
+
+   std::vector<double> wr= moldata3D.GetRelativeRcov(mol);
+
+   VectorXd Wr = getEigenVect(wr);
+
+   MatrixXd Bi;
+   MatrixXd tmp;
+   double TDBmat[8][10];
+   double dtmp;
+
+  for (int i=0;i<10;i++){
+    double * Bimat = GetGeodesicMatrix(dist, i+1, numAtoms);
+    Map<MatrixXd> Bi(Bimat, numAtoms,numAtoms);
+    MatrixXd RBi=Bi.cwiseProduct(dm);
+    double Bicount=(double) Bi.sum();
+
+    tmp=Wu.transpose() * RBi * Wu / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[0][i]=dtmp;
+
+    tmp=Wm.transpose() * RBi * Wm / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[1][i]=dtmp;
+
+    tmp=Wv.transpose() * RBi * Wv / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[2][i]=dtmp;
+
+    tmp=We.transpose() * RBi * We / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[3][i]=dtmp;
+
+    tmp=Wp.transpose() * RBi * Wp / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[4][i]=dtmp;
+
+    tmp=Wi.transpose() * RBi * Wi / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[5][i]=dtmp;
+
+    tmp=Ws.transpose() * RBi * Ws / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[6][i]=dtmp;
+
+    tmp=Wr.transpose() * RBi * Wr / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[7][i]=dtmp;
+
+  }
+
+  // create the Output vector!
+    w= AppendDouble(w, TDBmat[0], 10, 0);
+    w= AppendDouble(w, TDBmat[1], 10, 10);
+    w= AppendDouble(w, TDBmat[2], 10, 20);
+    w= AppendDouble(w, TDBmat[3], 10, 30);
+    w= AppendDouble(w, TDBmat[4], 10, 40);
+    w= AppendDouble(w, TDBmat[5], 10, 50);
+    w= AppendDouble(w, TDBmat[6], 10, 60);
+    w= AppendDouble(w, TDBmat[7], 10, 70);
```

Comment:
is the assignment to w necessary? This flattens the matrix into an array, doesn't it? Why not use a loop?